tests/by-util/test_fmt.rs

```diff
@@ -172,10 +187,42 @@ fn test_fmt_invalid_goal() {
             .args(&["one-word-per-line.txt", param, "invalid"])
             .fails()
             .code_is(1)
-            .stderr_contains("invalid value 'invalid'");
+            // GNU complains about "invalid width", which is confusing.
+            // We intentionally deviate from GNU, and show a more helpful message:
+            .stderr_contains("invalid goal: 'invalid'");
     }
 }
 
+#[test]
+fn test_fmt_invalid_goal_override() {
+    new_ucmd!()
+        .args(&["one-word-per-line.txt", "-g", "apple", "-g", "74"])
+        .succeeds()
+        .stdout_is("this is a file with one word per line\n");
+}
+
+#[test]
+fn test_fmt_invalid_goal_width_priority() {
+    new_ucmd!()
+        .args(&["one-word-per-line.txt", "-g", "apple", "-w", "banana"])
+        .fails()
+        .code_is(1)
+        .no_stdout()
+        .stderr_is("fmt: invalid width: 'banana'\n");
+    new_ucmd!()
+        .args(&["one-word-per-line.txt", "-w", "apple", "-g", "banana"])
+        .fails()
+        .code_is(1)
+        .no_stdout()
+        .stderr_is("fmt: invalid width: 'apple'\n");
```

Comment:
This test is a bit confusing: while reading I wondered why you swapped the options (from `-g -w` to `-w -g`) but kept the values at the original positions. I think this distracts from what the test function is about: to show (and ensure) that an invalid width has a higher priority than an invalid goal.

My suggestion is to remove this test because the next test (`-w banana -g apple`) tests the same but in a more intuitive way.